config/ompi_check_libhcoll.m4

```diff
@@ -47,6 +47,15 @@ AC_DEFUN([OMPI_CHECK_HCOLL],[
                               [ompi_check_hcoll_happy="yes"],
                               [ompi_check_hcoll_happy="no"])
 
+           AS_IF([test "$ompi_check_hcoll_happy" = "yes"],
+                 [
+                     CPPFLAGS=$coll_hcoll_CPPFLAGS
+                     LDFLAGS=$coll_hcoll_LDFLAGS
+                     LIBS=$coll_hcoll_LIBS
+                     AC_CHECK_FUNCS(hcoll_context_free, [], [])
+                 ],
+                 [])
```

Comment:
Minor style nit: you don't need to specify the empty 3rd argument.  I.e., you can just end the AS_IF with ]) on line 56 (and not have the additional `[])` block on line 57.